kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigConfigMap.java

```diff
@@ -141,27 +143,38 @@ private static void addAcceptedConditions(VirtualKafkaClusterStatusFactory statu
 
     private static void addResolvedRefsConditions(VirtualKafkaClusterStatusFactory statusFactory, ProxyModel proxyModel, ProxyConfigData data) {
         proxyModel.resolutionResult().clusterResults().stream()
-                .filter(ResolutionResult.ClusterResolutionResult::isAnyDependencyUnresolved)
+                .filter(ResolutionResult.ClusterResolutionResult::hasAnyResolutionProblems)
                 .forEach(clusterResolutionResult -> {
-
-                    Comparator<LocalRef<?>> comparator = Comparator.<LocalRef<?>, String> comparing(LocalRef::getGroup)
-                            .thenComparing(LocalRef::getKind)
-                            .thenComparing(LocalRef::getName);
-
-                    LocalRef<?> firstUnresolvedDependency = clusterResolutionResult.unresolvedDependencySet().stream()
-                            .sorted(comparator).findFirst()
-                            .orElseThrow();
-                    String message = String.format("Resource %s was not found.",
-                            ResourcesUtil.namespacedSlug(firstUnresolvedDependency, clusterResolutionResult.cluster()));
                     VirtualKafkaCluster cluster = clusterResolutionResult.cluster();
-
+                    VirtualKafkaCluster patch;
+                    ResolutionResult.UnresolvedReferences unresolvedReferences = clusterResolutionResult.unresolvedReferences();
+                    if (!unresolvedReferences.unresolved().isEmpty()) {
```

Comment:
here we've departed from a pure refactor, to support the VKC side I added the responsibility for finding ResolvedRefs: False conditions to the DependencyResolver, which meant the KafkaProxyReconciler side also needs to tolerate this case.